### PR TITLE
Add unit tests to hmda microservice

### PR DESCRIPTION
**Pull request by jaredb96**

### CODE REVIEW

File: hmda/src/test/scala/hmda/api/http/admin/InstitutionAdminHttpApiSpec.scala [hunk 1]
```diff
@@ -131,6 +180,25 @@ class InstitutionAdminHttpApiSpec extends AkkaCassandraPersistenceSpec with Must
       }
     }
 
+    "Create institution when it doesn't exist for the lei" in {
+      val testSampleInstitution =
```
Comment (rkovalik-raft) [inline hmda/src/test/scala/hmda/api/http/admin/InstitutionAdminHttpApiSpec.scala:184]: Same as my above comment, I think you can just copy teh sampleInstitution and use copy() to make a change to one field so it is a new institution.

File: hmda/src/test/scala/hmda/api/http/codec/filing/submission/EditDetailsSummaryCodecSpec.scala [hunk 2]
```diff
@@ -15,6 +15,7 @@ class EditDetailsSummaryCodecSpec
     forAll(editDetailsSummaryGen) { editDetailsSummary =>
       whenever(!editDetailsSummary.isEmpty) {
         val json = editDetailsSummary.asJson
+        val x = 5
```
Comment (rkovalik-raft) [inline hmda/src/test/scala/hmda/api/http/codec/filing/submission/EditDetailsSummaryCodecSpec.scala:18]: What is this for?

File: hmda/src/test/scala/hmda/api/http/model/StatsSpec.scala [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package hmda.api.http.model
+
+import org.scalatest.{MustMatchers, WordSpec}
+import hmda.api.http.model.{ SignedLeiCountResponse, SignedLeiListResponse, TotalLeiCountResponse, TotalLeiListResponse }
+
+class StatsSpec extends WordSpec with MustMatchers {
+  "response should have correct count" in {
+    val response = SignedLeiCountResponse(3)
+    assert(SignedLeiCountResponse.apply(3) == response)
```
Comment (rkovalik-raft) [inline hmda/src/test/scala/hmda/api/http/model/StatsSpec.scala:9]: This feels like testing if 0.0 = 0 to me. Are there better unit tests we can right for this?

File: hmda/src/test/scala/hmda/api/http/admin/InstitutionAdminHttpApiSpec.scala (hunk 1 above)
Comment (jaredb96) [reply hmda/src/test/scala/hmda/api/http/admin/InstitutionAdminHttpApiSpec.scala]: Agreed, fixed!

File: hmda/src/test/scala/hmda/api/http/admin/InstitutionAdminHttpApiSpec.scala [hunk 4]
```diff
@@ -181,15 +164,10 @@ class InstitutionAdminHttpApiSpec extends AkkaCassandraPersistenceSpec with Must
...
 
     "Create institution when it doesn't exist for the lei" in {
-      val testSampleInstitution =
-        institutionGen.sample
-          .getOrElse(Institution.empty)
-          .copy(LEI = lei)
-          .copy(taxId = Option("12-3456789"))
-          .copy(activityYear = 2019)
-      Put("/institutions", testSampleInstitution) ~> institutionAdminRoutes(oAuth2Authorization) ~> check {
+      val sampleEmptyInstitution = sampleInstitution.copy(activityYear = 2019)
```
Comment (rkovalik-raft) [inline hmda/src/test/scala/hmda/api/http/admin/InstitutionAdminHttpApiSpec.scala:167]: Nit: Maybe rename to `sampleNewInstitution`? Empty doesn't make a lot of sense to me

Comment (jaredb96) [reply hmda/src/test/scala/hmda/api/http/admin/InstitutionAdminHttpApiSpec.scala]: Thats true, change it

**Review by rkovalik-raft — APPROVED** [review]

**Pull request merged**